app/controllers/spree/admin/invoices_controller.rb

```diff
@@ -6,13 +6,34 @@ class InvoicesController < Spree::Admin::BaseController
       respond_to :json
       authorize_resource class: false
 
+      def index
+        @order = Spree::Order.find_by_number(params[:order_id])
+      end
+
       def create
         invoice_service = BulkInvoiceService.new
         invoice_service.start_pdf_job(params[:order_ids])
 
         render json: invoice_service.id, status: :ok
       end
 
+      def generate
+        @order = Order.find_by_number(params[:order_id])
+        if @order.can_generate_new_invoice?
+          @order.invoices.create!({
+                                    date: Time.now.to_date,
+                                    number: @order.next_invoice_number,
+                                    data: invoice_data
+                                  })
+        elsif @order.can_update_latest_invoice?
+          @order.invoices.last.update!({
+                                          date: Time.now.to_date,
+                                          data: invoice_data
+                                        })
```

Comment:
Since another PR is just working on improving layout and line length, I would try to reduce the indent which also means that you don't have to change it when the method call name is changed. And it's also recommended to always use `Time.zone`. And there's a method `today`:


```suggestion
          @order.invoices.create!(
            date: Time.zone.today,
            number: @order.next_invoice_number,
            data: invoice_data
          )
        elsif @order.can_update_latest_invoice?
          @order.invoices.last.update!(
            date: Time.zone.today,
            data: invoice_data
          )
```